introduction/A_overview.md

```diff
@@ -0,0 +1,101 @@
+
+The aim of this introductory guide is to present a brief, high level overview of Phoenix, the parts that make it up, and the layers underneath that support it.
+
+### Phoenix
+
+Phoenix is actually the top layer of a multi-layer system designed to be modular and flexible. The other layers are the Elixir middleware project, Plug, and the Erlang web server, Cowboy. Plug and Cowboy are covered in the next sections of this guide.
+
+Phoenix is made up of a number of distinct parts, each with its own purpose and role to play in building a web application. We will cover them all in depth throughout these guides, but here's a quick breakdown.
+
+- The Endpoint
+- handles all aspects of requests up until the beginning of our applications
+- provides a core set of middleware to apply to all requests
+- dispatches requests into designated routers
+- The Router
+- parses incoming requests and dispatches to the correct controller/action, passing parameters as needed
+- provides helpers to generate route paths or urls to resources
+- Pipelines
+- Allow easy segregation of middleware across different routes
+- Controllers
+- provide functions, called *actions*, to handle requests
+- Actions
+- prepare data and pass it into views
+- invoke rendering via views
+- perform redirects
+- Views
+- render templates
+- act as a presentation layer
+- define helper functions, available in templates, to decorate raw data
+- Templates
+- are what they sound like :)
+- Precompiled & fast
+- Channels
+- manage sockets for easy realtime communication
+- analogous to controllers except that they allow bi-directional communication with persistent connections
+- PubSub
+- Underlies the channel layer and allows *topics* to be subscribed to
+- Abstracts the underlying pubsub adapter for third-party pubsub integration
+
+### Plug
+
+[Plug](http://hexdocs.pm/plug/) is Elixir's web middleware layer. Conceptually, it shares a lot with other middleware layers like Rack for Ruby or WSGI for Python. Plugs are reusable modules that share the same very small, very regular public api. They provide discrete behaviors - like request header parsing or logging. Because the Plug api is so consistent, they can be stacked and executed in a set order, like a pipeline. They can also be re-used within a project or across projects.
+
+Plugs can be written to handle almost anything, from authentication to parameter pre-processing, and even rendering.
+
+Phoenix takes great advantage of Plug in general - the router and controllers especially so.
+
+One of the most important things about Plug, is that it provides adapters to HTTP servers which will ultimately deliver application content to your users. Currently, Plug only provides an adapter to Cowboy, which we will talk about next, but there are plans to provide adapters for other servers in the future.
+
+Links to more in-depth information on Plug can be found in the [Resources Guide](http://www.phoenixframework.org/docs/resources).
```

Comment:
What do you think about linking to the specific section of the page ? (i.e. http://www.phoenixframework.org/docs/resources#section-plug). The same for the link of cowboy below.